invenio_drafts_resources/services/records/components/files.py

```diff
@@ -36,23 +41,24 @@ def create(self, identity, data=None, record=None, errors=None):
               (this interface is used in records-resources and rdm-records)
         """
         draft = record
-        enabled = data.get("files", {}).get("enabled", True)
+        files = self.get_record_files(draft)
+        enabled = data.get(self.files_data_key, {}).get("enabled", True)
 
-        if draft.files.enabled != enabled:
+        if files.enabled != enabled:
```

Comment:
can we join the nested `if`s? The return only happens inside the inner if, so I _think_ the behaviour is left the same.
```suggestion
        if files.enabled != enabled and not self.service.check_permission(
                 identity, "manage_files", record=draft
             ):
```